compiler/hash-typecheck/src/diagnostics/error.rs

```diff
@@ -1189,46 +1189,50 @@ impl<'tc> From<TcErrorWithStorage<'tc>> for Report {
             TcError::TraitImplMissingMember {
                 trt_impl_term_id,
                 trt_def_term_id,
-                trt_def_missing_member_term_id,
+                missing_trt_members,
             } => {
+                let TrtDef { members, .. } =
+                    ctx.oracle().term_as_trt_def(*trt_def_term_id).expect("trait def term");
+                let trt_scope = ctx.reader().get_scope_copy(members);
+
+                let missing = missing_trt_members
+                    .iter()
+                    .map(|index| trt_scope.get_by_index(*index).name())
+                    .collect_vec();
+
+                // Create a sequence display for displaying member names
+                let missing_members = SequenceDisplay::new(
+                    &missing,
+                    SequenceDisplayOptions::with_limit(SequenceJoinMode::All, 6),
+                );
+
                 builder.with_error_code(HashErrorCode::TraitImplMissingMember).with_message(
                     format!(
-                        "trait `{}` is missing the member `{}`",
+                        "trait `{}` is missing the trait item{} {missing_members}",
```

Comment:
missing space? Also maybe "member" instead of "trait item"?

Reply:
No missing space, `pluralise!` basically converts the number into a suffix for a given word, which means that `1` yields nothing and anything greater yields `s`.

Reply:
the two phrases are synonymous so I don't mind which one we use

Reply:
Probably member is best for the sake of consistency